modules/aws-databricks-unity-catalog/uc_cross_account_role.tf

```diff
@@ -51,6 +51,15 @@ resource "aws_iam_policy" "unity_metastore" {
           "${aws_s3_bucket.metastore.arn}/*"
         ],
         "Effect" : "Allow"
+       },
+      {
+        "Action" : [
+          "sts:AssumeRole"
+        ],
+        "Resource" : [
+          "arn:aws:iam::${var.databricks_account_id}:role/${var.prefix}-uc-access"
```

Comment:
have you tested that this would work. Do it need to be something similar to 
```
  statement {
    sid     = "ExplicitSelfRoleAssumption"
    effect  = "Allow"
    actions = ["sts:AssumeRole"]
    principals {
      type        = "AWS"
      identifiers = ["arn:aws:iam::${var.aws_account_id}:root"]
    }
    condition {
      test     = "ArnLike"
      variable = "aws:PrincipalArn"
      values   = ["arn:aws:iam::${var.aws_account_id}:role/${var.prefix}-uc-access"]
    }
```

Reply:
The cloudformation quickstart setup also didn't require the ArnLike condition and just allowed for resource matching, so my guess is that this TF will also work without the ArnLike condition. Can you confirm @hectorcast-db?

Reply:
The condition one is for the trust policy, where you have a principal. For this type of policy, where you have resources and not principals, it is not needed.